gcp/website/frontend3/src/search.js

```diff
@@ -127,6 +144,223 @@ export class ExpandableSearch {
     container.toggle.classList.remove('active');
     container.toggle.setAttribute('aria-expanded', 'false');
     
-    if (container.input) container.input.blur();
+    if (container.input) {
+      container.input.blur();
+      // Hide suggestions when closing search
+      const suggestionsManager = this.suggestionsManagers.get(container.input);
+      if (suggestionsManager) {
+        suggestionsManager.hide();
+      }
+    }
+  }
+}
+
+// ============= Search Suggestions Manager =============
+
+export class SearchSuggestionsManager {
+  constructor(inputElement) {
+    this.input = inputElement;
+    this.suggestionsElement = null;
+    this.selectedIndex = -1;
+    this.currentSuggestions = [];
+    this.debounceTimer = null;
+    this.isDestroyed = false;
+    
+    this.cleanupOrphanedElements();
+    
+    this.init();
+  }
+
+  init() {
+    this.createSuggestionsElement();
+    this.setupEventListeners();
+  }
+
+  cleanupOrphanedElements() {
+    const orphanedElements = document.querySelectorAll('.search-suggestions');
+    orphanedElements.forEach(element => {
```

Comment:
Will this cause issues if we have two search boxes on the page? (on the list page we have the main one, and the one in the top left corner that you added earlier.)

Reply:
Wait, this is just checking whether managerId exists, wouldn't that always be the case, since it's added every time you create it, and I don't really see anywhere you remove this.